phpBB/phpbb/install/helper/config.php

```diff
@@ -0,0 +1,386 @@
+	 * 								does not exist.
+	 *
+	 * @return 	mixed	value of the specified parameter or the default value if the data
+	 * 					cannot be recovered.
+	 */
+	public function get($param_name, $default = false)
+	{
+		return (isset($this->installer_config[$param_name])) ? $this->installer_config[$param_name] : $default;
+	}
+
+	/**
+	 * Sets a parameter in installer_config
+	 *
+	 * @param	string	$param_name	Name of the parameter
+	 * @param	mixed	$value		Values to set the parameter
+	 */
+	public function set($param_name, $value)
+	{
+		$this->installer_config = array_merge($this->installer_config, array(
+			$param_name => $value,
+		));
+	}
+
+	/**
+	 * Returns system parameter
+	 *
+	 * @param string	$param_name	Name of the parameter
+	 *
+	 * @return mixed	Returns system parameter if it is defined, false otherwise
+	 */
+	public function system_get($param_name)
+	{
+		return (isset($this->system_data[$param_name])) ? $this->system_data[$param_name] : false;
+	}
+
+	/**
+	 * Returns remaining time until the run time limit
+	 *
+	 * @return int	Remaining time until the run time limit in seconds
+	 */
+	public function get_time_remaining()
+	{
+		if ($this->system_data['max_execution_time'] <= 0)
+		{
+			return 1;
+		}
+
+		return ($this->system_data['start_time'] + $this->system_data['max_execution_time']) - time();
+	}
+
+	/**
+	 * Returns remaining memory available for PHP
+	 *
+	 * @return int	Remaining memory until reaching the limit
+	 */
+	public function get_memory_remaining()
+	{
+		if ($this->system_data['memory_limit'] <= 0)
+		{
+			return 1;
+		}
+
+		if (function_exists('memory_get_usage'))
+		{
+			return ($this->system_data['memory_limit'] - memory_get_usage());
+		}
+
+		// If we cannot get the information then just return a positive number (and cross fingers)
+		return 1;
+	}
+
+	/**
+	 * Saves the latest executed task
+	 *
+	 * @param string	$task_service_name	Name of the installer task service
+	 */
+	public function set_finished_task($task_service_name)
+	{
+		$this->progress_data['last_task_name']	= $task_service_name;
+	}
+
+	/**
+	 * Set active module
+	 *
+	 * @param string	$module_service_name	Name of the installer module service
+	 */
+	public function set_active_module($module_service_name)
+	{
+		$this->progress_data['last_task_module_name']	= $module_service_name;
+	}
+
+	/**
+	 * Getter for progress data
+	 *
+	 * @return array
+	 */
+	public function get_progress_data()
+	{
+		return $this->progress_data;
+	}
+
+	/**
+	 * Recovers install configuration from file
+	 */
+	public function load_config()
+	{
+		if (!$this->filesystem->exists($this->install_config_file))
+		{
+			return;
+		}
+
+		$file_content = @file_get_contents($this->install_config_file);
+		$serialized_data = trim(substr($file_content, 8));
+
+		$this->installer_config = array();
+		$this->progress_data = array();
+		$this->navigation_data = array();
+
+		if (!empty($serialized_data))
+		{
+			$unserialized_data = unserialize($serialized_data);
```

Comment:
I'd prefer it if we would be using the json format instead of serialize(). There are definitely some implications and possible issues with using serialize() and unserialize()